src/pybamm/expression_tree/operations/serialise.py

```diff
@@ -237,6 +241,228 @@ def load_model(
             """
         )
 
+    @staticmethod
+    def _json_encoder(obj):
+        if isinstance(obj, Enum):
+            return obj.name
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.integer):
+            return int(obj)
+        else:
+            raise TypeError(f"Object of type {type(obj)} is not JSON serializable.")
+
+    @staticmethod
+    def save_custom_model(model, filename=None):
+        """
+        Saves a custom (non-discretised) PyBaMM model to a JSON file. Works for user defined models that are subclasses of BaseModel.
+
+        This includes symbolic expressions for rhs, algebraic, initial and boundary
+        conditions, events, and variables. Useful for storing or sharing models
+        before discretisation.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The custom symbolic model to be saved.
+        filename : str, optional
+            The desired name of the JSON file. If not provided, a name will be
+            generated from the model name and current datetime.
+
+        Example
+        -------
+        >>> import pybamm
+        >>> model = pybamm.lithium_ion.BasicDFN()
+        >>> from pybamm.expression_tree.operations.serialise import Serialise
+        >>> Serialise.save_custom_model(model, "basicdfn_model")
+
+        """
+        SCHEMA_VERSION = "1.0"
+        model_json = {
+            "pybamm_version": pybamm.__version__,
+            "schema_version": SCHEMA_VERSION,
```

Comment:
Nice, thanks for adding this. Could you add two tests for this?
- a test that loads a JSON file with an unsupported schema version and checks that a `ValueError` is raised
- a test that checks that that the saved JSON files do contain the schema version